### DISCUSSION TITLE
Error in the save mode: Would you mind helping me

### DISCUSSION THREAD

**Discussion by Brayan532** [ultralytics/ultralytics#16244] [Q&A] [answered]
Hi there, 
I have made the following change in the yolov8, 

# YOLOv8.0n backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]] # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]] # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]] # 5-P4/16
  - [-1, 6, C2f, [512, True]]
## the below part was added  
- [-1, 1, Conv, [512, 3, 1]]
## 
  - [-1, 1, Conv, [1024, 3, 2]] # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 9

Then, switched to train the model. 
The first part of train as you see passed without any problem, however, then I got an error for which I need your help as I've been stuck and am not able to continue. 

Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/2         0G       3.61      5.567      4.267        216        640: 100%|██████████| 50/50 [05:55<00:00,  7.12s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 25/25 [01:31<00:00,  3.66s/it]                   all        800       5620          0          0          0          0

Traceback (most recent call last):

  File ~\anaconda\envs\od1\lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\spec\documents\code\test.py:10
    model.train(data="data.yaml", epochs=2, imgsz=640)

  File ~\anaconda\envs\od1\lib\site-packages\ultralytics\engine\model.py:811 in train
    self.trainer.train()

  File ~\anaconda\envs\od1\lib\site-packages\ultralytics\engine\trainer.py:206 in train
    self._do_train(world_size)

  File ~\anaconda\envs\od1\lib\site-packages\ultralytics\engine\trainer.py:439 in _do_train
    self.save_model()

  File ~\anaconda\envs\od1\lib\site-packages\ultralytics\engine\trainer.py:487 in save_model
    torch.save(

  File ~\anaconda\envs\od1\lib\site-packages\ultralytics\utils\patches.py:110 in torch_save
    return _torch_save(*args, **kwargs)

  File ~\anaconda\envs\od1\lib\site-packages\torch\serialization.py:652 in save
    _save(obj, opened_zipfile, pickle_module, pickle_protocol, _disable_byteorder_record)

  File ~\anaconda\envs\od1\lib\site-packages\torch\serialization.py:864 in _save
    pickler.dump(obj)

PicklingError: Can't pickle <class 'ultralytics.nn.tasks.DetectionModel'>: it's not the same object as ultralytics.nn.tasks.DetectionModel

**Comment by glenn-jocher** [comment]
@Brayan532 it seems like the error might be related to changes in the model architecture affecting serialization. Try reverting the changes to see if the issue persists. If the problem is resolved, consider reviewing the modifications for compatibility with the existing framework. For further assistance, you can refer to the [YOLO Common Issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/).

**Reply by Brayan532** [reply]
> @Brayan532 it seems like the error might be related to changes in the model architecture affecting serialization. Try reverting the changes to see if the issue persists. If the problem is resolved, consider reviewing the modifications for compatibility with the existing framework. For further assistance, you can refer to the [YOLO Common Issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/).

Dear Glenn, 
Thank you for your reply. I have taken you advice and double checked everything. I have reverted the only change made in yolov8.yaml. In my PC, I’ve got the same error. I've noticed that the problem is solved if I increase the number of epochs, as it was initially 2 epochs, I have no clue about this, however, you might know why when it was increased to 20, no error pops up. Best wishes.

**Reply by glenn-jocher** [reply, marked as answer]
@Brayan532 it seems the issue might be related to the model not having enough time to stabilize with only 2 epochs. Increasing the epochs allows the model to train more thoroughly, which might resolve serialization issues. If you need further guidance, feel free to ask!
